admin/bin/qserv-admin.py

```diff
@@ -340,9 +337,79 @@ def _parseShow(self, tokens):
         t = tokens[0].upper()
         if t == 'DATABASES':
             self._impl.showDatabases()
+        elif t == 'NODES':
+            self._parseShowNodes()
         else:
             raise QservAdminException(QservAdminException.BAD_CMD)
 
+    def _parseShowNodes(self):
+        """
+        Subparser, handle all SHOW NODES requests.
+        """
+
+        nodes = self._impl.getNodes()
+        if not nodes:
+            print "No nodes defined in CSS"
+        for node in sorted(nodes.keys()):
+            options = sorted(nodes[node].items())
+            options = ["{0}={1}".format(k, v) for k, v in options]
+            print "{0} {1}".format(node, ' '.join(options))
+
+    def _parseUpdate(self, tokens):
+        """
+        Subparser - handles all UPDATE requests.
+        """
+        t = tokens[0].upper()
+        if t == 'NODE':
+            self._parseUpdateNode(tokens[1:])
+        else:
+            raise QservAdminException(QservAdminException.BAD_CMD)
+
+    def _parseUpdateNode(self, tokens):
+        """
+        Subparser - handles all UPDATE NODE requests.
+        Throws KvException, QservAdminException.
+        """
+        if len(tokens) < 2:
+            raise QservAdminException(QservAdminException.BAD_CMD,
+                                "Unexpected number of arguments.")
+
+        requiredKeys = self.requiredOpts['updateNode']
+
+        # parse key=value pairs
+        options = {'nodes': [tokens[0]]}
+
+        for key, value in self._parseKVList(tokens[1:], requiredKeys):
+            options[key] = value
+
+        # check that all required options are there
+        self._checkExist(options, requiredKeys)
+
+        # call CSS to do the rest, remap options to argument names
+        self._impl.setNodeState(**options)
+
+    def _parseKVList(self, tokens, possibleKeys=None):
+        """
+        Parse the series of key=value strings, returns the list of (key, value)
+        tuples. If possibleKeys is not None then keys are checked agains that
+        list, exception is raised if key is not in the list.
+        """
+        kvList = []
+        for opt in tokens:
+            if '=' not in opt:
+                raise QservAdminException(QservAdminException.BAD_CMD,
+                   "Invalid argument '%s', should be <key>=<value>" % opt)
+            key, value = opt.split('=', 1)
+
+            if possibleKeys is not None and key not in possibleKeys:
+                raise QservAdminException(QservAdminException.BAD_CMD,
+                   "Unrecognized option key '%s', possible keys: %s" % \
+                   (key, ' '.join(possibleKeys)))
+
+            kvList.append((key, value))
+
+        return kvList
+
```

Comment:
We may have to use a parser module if parsing will get more complex here? However, this would be done in an other ticket.